src/main/java/me/xmrvizzy/skyblocker/skyblock/item/PriceInfoTooltip.java

```diff
@@ -156,54 +183,108 @@ public static String getInternalNameFromNBT(ItemStack stack) {
+                if (ea.contains("petInfo")) {
+                    JsonObject petInfo = gson.fromJson(ea.getString("petInfo"), JsonObject.class);
+                    internalName += "-" + petInfo.get("type").getAsString() + "-" + petInfo.get("tier").getAsString();
+                }
+            } else if ("POTION".equals(internalName)) {
+                String enhanced = ea.contains("enhanced") ? "-ENHANCED" : "";
+                String extended = ea.contains("extended") ? "-EXTENDED" : "";
+                String splash = ea.contains("splash") ? "-SPLASH" : "";
+                if (ea.contains("potion") && ea.contains("potion_level")) {
+                    internalName += "-" + ea.getString("potion").toUpperCase() + "-" + ea.getInt("potion_level")
+                            + enhanced + extended + splash;
+                }
+            } else if ("RUNE".equals(internalName)) {
+                if (ea.contains("runes")) {
+                    NbtCompound runes = ea.getCompound("runes");
+                    String rune = runes.getKeys().stream().findFirst().get();
+                    internalName += "-" + rune.toUpperCase() + "-" + runes.getInt(rune);
                 }
             }
+
         }
         return internalName;
     }
 
     private static Text getCoinsMessage(double price, int count) {
         if (count == 1) {
-            String priceString = String.format(Locale.ENGLISH, "%1$,.0f", price);
+            String priceString = String.format(Locale.ENGLISH, "%1$,.1f", price);
             return new LiteralText(priceString + " Coins").formatted(Formatting.DARK_AQUA);
         } else {
-            String priceString = String.format(Locale.ENGLISH, "%1$,.0f", price * count);
+            String priceString = String.format(Locale.ENGLISH, "%1$,.1f", price * count);
             LiteralText priceText = (LiteralText) new LiteralText(priceString + " Coins ").formatted(Formatting.DARK_AQUA);
-            priceString = String.format(Locale.ENGLISH, "%1$,.0f", price);
+            priceString = String.format(Locale.ENGLISH, "%1$,.1f", price);
             LiteralText priceText2 = (LiteralText)  new LiteralText( "(" + priceString + " each)").formatted(Formatting.GRAY);
             return priceText.append(priceText2);
         }
     }
 
-    public static boolean firstRun = true;
+    public static int minute = 0;
     public static void init() {
-        TimerTask repeatedTask = new TimerTask() {
-            public void run() {
-                if (SkyblockerConfig.get().general.itemTooltip.enableAvgBIN || PriceInfoTooltip.firstRun)
-                    CompletableFuture.runAsync(PriceInfoTooltip::downloadAvgPrices);
-                if (SkyblockerConfig.get().general.itemTooltip.enableLowestBIN || PriceInfoTooltip.firstRun)
+        skyblocker.scheduler.scheduleCyclic(() -> {
+            {
+                if ((SkyblockerConfig.get().general.itemTooltip.enableAvgBIN || minute == 0)
```

Comment:
If feature requires getting something from the internet and is disabled it shouldn't be creating that connection. Same goes for all other tooltips.